src/gazebo2rviz/gazebo2moveit_collision_node.py

```diff
@@ -31,12 +36,17 @@ def on_model_states_msg(model_states_msg):
     lastUpdateTime = rospy.get_rostime()
 
     for (model_idx, modelinstance_name) in enumerate(model_states_msg.name):
+        # Get the model name
         model_name = pysdf.name2modelname(modelinstance_name)
- 
-        if not modelinstance_name in model_cache:
+        # Get the object's type by a lookup in the gazebo mapping manager
+        if get_object_type_service is not None:
+            response = get_object_type_service(modelinstance_name)
+            if response.success:
+                model_name = pysdf.name2modelname(response.object_type)
```

Comment:
if the service returns object type, why are we assigning it as a model name. This code is unclear to me.